Services/data/devicemodels/scripts/SetFuelLevel-method.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+/*global log*/
+/*global updateState*/
+/*global updateProperty*/
+/*global sleep*/
+/*jslint node: true*/
+
+"use strict";
+
+/**
+ * Entry point function called by the simulation engine.
+ *
+ * @param context        The context contains current time, device model and id
+ * @param previousState  The device state since the last iteration
+ * @param previousProperties  The device properties since the last iteration
+ */
+
+/*jslint unparam: true*/
+function main(context, previousState, previousProperties) {
+
+    // General info indicating that this method is being called
+    log("Executing JavaScript SetFuelLevel method.");
+
+    // Input validation. Make sure that thew new
+    // fuel level is within the allowable range
+    var newFuelLevel = context.FuelLevel;
+    if (isNaN(newFuelLevel)) {
+        newFuelLevel = 0;
+    }
+    newFuelLevel = Math.max(newFuelLevel, 0);
```

Comment:
why do you need this check?  doesn't it get immediately overwritten by this:
   newFuelLevel = Math.min(newFuelLevel, 70);
no?

If you're ensuring it's between 0 & 70 why not write an if statement that keeps it in bounds?  the code as written doesn't seem as clear as it could be.

Reply:
Changed to use an if statement.